lib/saml/auth_fail_handler.rb

```diff
@@ -17,15 +19,17 @@ def initialize(saml_response, user, session)
       @context = nil
     end
 
-    def known_error?
-      known_errors = [
-        :clicked_deny?,
-        :auth_too_late?,
-        :auth_too_early?
-      ]
+    def error
+      KNOWN_ERRORS.each do |known_error|
+        return known_error if send("#{known_error}?")
+      end
```

Comment:
@saneshark - does this just apply to `auth_too_late?` and `auth_too_early?` messages from `@saml_response.errors[0]`, or is there a way to apply it to the `clicked_deny?` method, too?